_posts/2016-05-25-gwf-workshop.md

```diff
@@ -0,0 +1,994 @@
+  the_geom_webmercator 
+FROM 
+  ne_50m_land
+```
+
+#### Adding **World Robinson** projection (ESPG:54030):
+
+```sql
+INSERT INTO spatial_ref_sys
+  (srid, auth_name, auth_srid, proj4text, srtext) values (54030, 'EPSG', 54030,
+'+proj=robin +datum=WGS84','PROJCS["World_Robinson",
+    GEOGCS["GCS_WGS_1984",
+        DATUM["WGS_1984",
+            SPHEROID["WGS_1984",6378137,298.257223563]],
+        PRIMEM["Greenwich",0],
+        UNIT["Degree",0.017453292519943295]],
+    PROJECTION["Robinson"],
+    PARAMETER["False_Easting",0],
+    PARAMETER["False_Northing",0],
+    PARAMETER["Central_Meridian",0],
+    UNIT["Meter",1],
+    AUTHORITY["EPSG","54030"]]');
+```
+
+```sql
+SELECT 
+  cartodb_id, ST_Transform(the_geom, 54030) AS the_geom_webmercator
+FROM 
+  ne_50m_land
+```
+
+![robinson](../img/160520-zgz/robinson.png)
+
+_About [`ST_Transform`](http://postgis.net/docs/ST_Transform.html)._
+
+### 4.3 Geoprocessing
+
+#### Create a **buffer** from points:
+
+```sql
+SELECT
+  cartodb_id,
+  name,
+  ST_Transform(
+    ST_Buffer(the_geom::geography, 50000)::geometry
+    ,3857
+  ) AS the_geom_webmercator
+FROM
+  populated_places
+WHERE
+  name ilike 'madrid'
+```
+
+![buffer](../img/160520-zgz/buffer.png)
+
+_About [`ST_Buffer`](http://postgis.net/docs/ST_Buffer.html)._
+
+#### Get the **difference** between two geometries:
+
+```sql
+SELECT
+  a.cartodb_id,
+    ST_Difference(
+        a.the_geom_webmercator,
+        b.the_geom_webmercator
+  ) AS the_geom_webmercator
+FROM
+  ne_50m_land a,
+  ne_adm0_europe b
+WHERE
+  b.adm0_a3 like 'ESP'
+```
+
+![difference](../img/160520-zgz/difference.png)
+
+_About [`ST_Difference`](http://postgis.net/docs/ST_Difference.html)._
+
+#### Get the number of points inside a polygon
+
+Using `GROUP BY`:
+
+```sql
+SELECT
+  b.cartodb_id,
+  b.name,
+  b.the_geom_webmercator,
+  count(*) AS pp_count,
+  sum(a.pop_max) as sum_pop
+FROM
+  ne_10m_populated_places_simple a,
+  ne_adm0_europe b
+WHERE
+  ST_Intersects(a.the_geom, b.the_geom)
+GROUP BY
+  b.cartodb_id,
+  b.name,
+  b.the_geom_webmercator
+```
+
+Using `LATERAL`:
+
+```sql
+SELECT
+  a.cartodb_id,
+  a.name,
+  a.the_geom_webmercator,
+  counts.number_cities,
+  to_char(counts.sum_pop,'999,999,999') as sum_pop --decimal separator
+FROM
+  ne_adm0_europe a
+CROSS JOIN LATERAL
+  (
+    SELECT
+      count(*) as number_cities,
+      sum(pop_max) as sum_pop
+    FROM
+      ne_10m_populated_places_simple b
+    WHERE
+      ST_Intersects(a.the_geom, b.the_geom)
+  ) AS counts
+```
+_About [`ST_Intersects`](http://postgis.net/docs/ST_Intersects.html) and [Lateral JOIN](http://blog.heapanalytics.com/postgresqls-powerful-new-join-type-lateral)_
+
+![ADD IMAGE](../img/)
+
+#### Know wether a geometry is **within** the given range from another geometry:
+
+```sql
+SELECT
+  a.*
+FROM
+  ne_10m_populated_places_simple a,
+  ne_10m_populated_places_simple b
+WHERE
+    a.cartodb_id != b.cartodb_id
+  AND ST_DWithin(
+      a.the_geom_webmercator,
```

Comment:
...as long as you are aware that those CRS units are not meters and depend directly on the latitude of your data

No need to complicate the query since it's an exercise, but remember the audience that CRS units in web mercator maps are never equal to meters.